tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -316,12 +317,31 @@ private void CheckUnboundProbes()
                     return;
                 }
 
+                // Initialize these lists only when there is at least one unbound probe that becomes bound, to reduce unnecessary allocations.
+                List<NativeLineProbeDefinition> lineProbes = null;
+                List<ProbeDefinition> noLongerUnboundProbes = null;
+
                 foreach (var unboundProbe in _unboundProbes)
```

Comment:
Can `_unbpundProbes` change during iteration?

Reply:
Nope, it holds the `_instanceLock`.